### PR TITLE
Added Interactive Capability for fetchModule Gradle command.

### PR DESCRIPTION
**Pull request by smsunarto**
Added Interactive capability for 'gradlew createmodule" command and 'gradlew fetchModule'.

### Contains

**fetchModule**
1. Automatically ask for a module name if module name is not typed in command.

### How to test

`FORMAT: gradlew fetchmodule<ID>`
1. Try executing 'gradlew fetchmodule<ID>'
2. Try executing just 'gradlew fetchmodule' (!Without ID!)

### Outstanding before merging
- Edit the wikidoc.

### Could be improved
- Multiple module fetch.

### CODE REVIEW

**Comment by GooeyHub** [comment]
Can one of the admins please verify this patch?

**Comment by Cervator** [comment]
Thanks :-)

It may take a little while to get around to testing this, same as the other similar PR.

@GooeyHub: ok to test

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: config/gradle/utility.gradle [hunk 1]
```diff
@@ -22,6 +22,15 @@ ext {
     templatesDir = 'templates'
 }
 
+// Prompts the user to enter a line of text
+def getUserString (String prompt) {
+    println ('\n' + (char)27 + '[1m*** ' + prompt + '\n') // Bold terminal text
```
Comment (msteiger) [inline config/gradle/utility.gradle]: Looks nice in bash - does it also work in other shells, and in particular on Windows command line?
When in doubt, I suggest not doing the bold text trick.

**Comment by smsunarto** [comment]
@msteiger tested this on windows, bold text works in windows cl.

**Comment by mjuvekar7** [comment]
The prompt which asks for the module name looks weird on my machine. I am using iTerm 2 on a macOS 10.12 machine. Heres a screenshot 
moduleselectprompt
Could be because of my weird bashrc and bash_profile.
Also, one little suggestion: could you ask for the module name **before** gradle does all the compiling/downloading stuff?

**Comment by flo** [comment]
createModule change is also covered by PR #2634

**Comment by jellysnake** [comment]
@smsunarto these changes appear to be a subset of those in #2634. Perhaps close this or alter it to enhance those not covered in the other pull?

**Comment by smsunarto** [comment]
@jellysnake closing this one, since I think @cervator is planning to do this through groovy instead of doing it through gradle.

**Comment by Cervator** [comment]
Can confirm I've grabbed this for a pure Groovy script, avoiding some of the Gradle quirks :-)

Thanks all! Getting closer ...

File: config/gradle/utility.gradle (hunk 1 above)
Comment (Cervator) [inline config/gradle/utility.gradle]: Checked this out as part of running it in a Groovy script on Windows either with regular `cmd` or inside the IntelliJ terminal but neither one worked :(

**Pull request closed without merging**